botsing-maven/src/main/java/eu/stamp/botsing/BotsingMojo.java

```diff
@@ -31,150 +37,233 @@
 @Mojo(name = "botsing", defaultPhase = LifecyclePhase.VERIFY, requiresDependencyResolution = ResolutionScope.TEST, threadSafe = true)
 public class BotsingMojo extends AbstractMojo {
 
-    /**
-     * To see all the properties available take a look at org.evosuite.Properties.java
-     */
-
-    /**
-     * Folder with dependencies to run the project
-     */
-    @Parameter(property = "project_cp")
-    private String projectCP;
-
-    /**
-     * Log file with the stacktrace
-     */
-    @Parameter(defaultValue = "sample.log", property = "crash_log")
-    private String crashLog;
-
-    /**
-     * The frame level up to which parse the stack trace
-     */
-    @Parameter(defaultValue = "3", property = "target_frame")
-    private Integer targetFrame;
-
-    /**
-     * Maven variables
-     */
-    @Parameter(defaultValue = "${project}", required = true, readonly = true)
-    private MavenProject project;
-
-    @Component
-    private RepositorySystem repoSystem;
-
-    @Parameter(defaultValue = "${repositorySystemSession}", readonly = true, required = true)
-    private RepositorySystemSession repoSession;
+	/**
+	 * To see all the properties available take a look at org.evosuite.Properties.java
+	 */
+
+	/**
+	 * Folder with dependencies to run the project
+	 */
+	@Parameter(property = "project_cp")
+	private String projectCP;
+
+	/**
+	 * Log file with the stacktrace
+	 */
+	@Parameter(defaultValue = "sample.log", property = "crash_log")
+	private String crashLog;
+
+	/**
+	 * The frame level up to which parse the stack trace
+	 */
+	@Parameter(defaultValue = "3", property = "target_frame")
+	private Integer targetFrame;
+
+	/**
+	* the size of the population that evolves during the search with a default value of 100
+	*/
+	@Parameter(defaultValue = "100", property = "population")
+	private Integer population;
+
+	/**
+	* the search budget in seconds with a default value of 1800
+	*/
+	@Parameter(defaultValue = "1800", property = "search_budget")
+	private Integer searchBudget;
+
+	/**
+	* the global timeout in seconds, after which the execution stops if the search is
+	* stuck with a default value of 1800 (the timeout is only reached if the search does not improve after 1800 seconds)
+	*/
+	@Parameter(defaultValue = "1800", property = "global_timeout")
+	private Integer globalTimeout;
+
+	/**
+	* the directory where the tests are generated with a default value of `crashreproduction-tests`
+	*/
+	@Parameter(defaultValue = "crashreproduction-tests", property = "test_dir")
+	private String testDir;
+
+	/**
+	* the seed used to initialize the random number generator. This value allows to have deterministic
+	* behavior and should be set when performing evaluations
+	*/
+	@Parameter(defaultValue = "100", property = "random_seed")
```

Comment:
The default value should be `null`. Otherwise, Botsing becomes deterministic. This value should be set by the user only for testing or experiment running purposes.

Reply:
removed default value